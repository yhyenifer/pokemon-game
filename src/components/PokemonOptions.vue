<template>
<div class="options-container">
  <ul>
    <li v-for="pokemon in this.pokemons"
        :key="pokemon.id"
        >
      <button @click="$emit('selectionPokemon', pokemon.id)">
        {{pokemon.name}}
      </button>
    </li>
  </ul>
</div>
</template>

<script>
export default {
  name: 'PokemonOptions',
  props: {
    pokemons: {
      type: [],
      required: true,
    },
  },
};
</script>

<style scoped>
/* Pokemon Options */
ul {
  list-style-type: none;
}
li {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  cursor: pointer;
  margin-bottom: 10px;
  width: 250px;
}
li button {
  width: 100%;
  background: white;
  border: 0;
  text-transform: capitalize;
  font-size: 24px;
}

li:hover {
  background-color: rgba(0, 0, 0, 0.05);
}

.options-container {
  display: flex;
  justify-content: center;
}
</style>
