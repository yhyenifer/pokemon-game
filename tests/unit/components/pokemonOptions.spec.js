import { shallowMount } from '@vue/test-utils';
import PokemonOptions from '@/components/PokemonOptions.vue';
import { pokemons } from '../mocks/pokemons.mock';

describe('PokemonOptions Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons,
      },
    });
  });

  it('debe de hacer match con el snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('debe mostrar 4 opciones correctamente', () => {
    const liTags = wrapper.findAll('li');
    expect(liTags.length).toBe(4);
    expect(liTags[0].text()).toBe('bulbasaur');
    expect(liTags[1].text()).toBe('ivysaur');
    expect(liTags[2].text()).toBe('venusaur');
    expect(liTags[3].text()).toBe('charmander');
  });

  it('debe de emitir "selection" con sus respectivos parÃ¡metros al hacer click', () => {
    const [b1, b2, b3, b4] = wrapper.findAll('button');

    b1.trigger('click');
    b2.trigger('click');
    b3.trigger('click');
    b4.trigger('click');
    // console.log(wrapper.emitted('selectionPokemon'))
    expect(wrapper.emitted('selectionPokemon').length).toBe(4);
    expect(wrapper.emitted('selectionPokemon')[0]).toEqual([1]);
    expect(wrapper.emitted('selectionPokemon')[1]).toEqual([2]);
    expect(wrapper.emitted('selectionPokemon')[2]).toEqual([3]);
    expect(wrapper.emitted('selectionPokemon')[3]).toEqual([4]);
  });
});
