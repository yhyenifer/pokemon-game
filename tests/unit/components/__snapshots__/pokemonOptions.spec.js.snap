// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component debe de hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul>
    <li><button>bulbasaur</button></li>
    <li><button>ivysaur</button></li>
    <li><button>venusaur</button></li>
    <li><button>charmander</button></li>
  </ul>
</div>
`;
